cpp/src/arrow/compute/api_scalar.h

```diff
@@ -316,6 +316,19 @@ struct ARROW_EXPORT CompareOptions {
   enum CompareOperator op;
 };
 
+enum BetweenOperator : int8_t {
+  BETWEEN_LESS_EQUAL_LESS_EQUAL,
+  BETWEEN_LESS_EQUAL_LESS_THAN,
+  BETWEEN_LESS_THAN_LESS_EQUAL,
+  BETWEEN_LESS_THAN_LESS_THAN,
+};
+
+struct ARROW_EXPORT BetweenOptions {
+  explicit BetweenOptions(BetweenOperator op) : op(op) {}
+  BetweenOptions() : BetweenOptions(BetweenOperator::BETWEEN_LESS_EQUAL_LESS_EQUAL) {}
+  enum BetweenOperator op;
```

Comment:
Four options seem to make sense, nesting further seems to complicate things unnecessarily

1. LESS_LESS
2. LESS_EQUAL_LESS
3. LESS_LESS_EQUAL
4. LESS_EQUAL_LESS_EQUAL

Reply:
Either use enum class, or do this:

```
class BetweenOptions {
    enum BetweenOperator {...};
};
```

That way the variants can be accessed as `BetweenOptions::LESS_LESS`. However, we shouldn't use a plain enum at the top level since the enum values won't be namespaced.

Reply:
Function options implemented.